portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -24,9 +34,74 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+  private ArrayList<String> messages;
+
+  public void init() {
+      //submitTime = new Date();
+      messages = new ArrayList<String>();
+  }
+
   @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+  public void doGet(HttpServletRequest request, HttpServletResponse response)
+  throws IOException {
+
+      Query query = new Query("Task//");//.addSort("timestamp", SortDirection.DESCENDING);
```

Comment:
ditto

Reply:
But also, do you have more info on this query? I'm not familiar with this format (genuine question, have no idea how this works).

Reply:
Not very clear, but I believe that you create a Query instance of a specific message entity that you want to store, then pass that query into the PreparedQuery object that uses datastore.prepare(query) method to store the comment in DataStore.